packages/vow/src/watch-utils.js

```diff
@@ -79,20 +85,67 @@ export const prepareWatchUtils = (
     },
     {
       utils: {
+        /** @param {EVow<unknown>[]} vows */
+        all(vows) {
+          return this.facets.helper.createVow(vows, false);
+        },
+        /** @param {EVow<unknown>[]} vows */
+        allSettled(vows) {
+          return this.facets.helper.createVow(vows, true);
+        },
```

Comment:
To implement the difference in return types you're looking for, since `Vow<any[]>` is the more general type, it could be as simple as narrowing the type within `allSettled`.

```suggestion
        /** @param {EVow<unknown>[]} vows */
        all(specimens) {
          return this.facets.helper.createVow(specimens, false);
        },
        /** @param {EVow<unknown>[]} specimens */
        allSettled(specimens) {
          /** @type {Vow<({status: 'fulfilled', value: any} | {status: 'rejected', reason: any})[]>} */
          const statuses = this.facets.helper.createVow(specimens, true);
          return statuses;
        },
```